src/Router.js

```diff
@@ -26,14 +25,17 @@ class App extends React.Component {
   };
 
   render() {
-    const AppNavigator = this.props.navigator;
+    const { navigator: AppNavigator, stateManager: { dispatch, getState } } = this.props;
+
     return (
-      <AppNavigator navigation={addNavigationHelpers({ dispatch: navigationStore.dispatch, state: navigationStore.state })} />
+      <AppNavigator navigation={addNavigationHelpers({ dispatch, state: getState(this.props) })} />
```

Comment:
`getState` as a function because in the `ReduxEngine` it needs the props so it can pull the nav state off of them. I am not sure if I can think of a better way around that. But this is "hidden complexity" that the end user never has to really care about.

Reply:
Hidden complexity is fine as long as it's in the engine, and not the router. The router shouldn't know what engine it's using.